src/main/java/htsjdk/samtools/BamFileIoUtils.java

```diff
@@ -20,10 +20,10 @@
 public class BamFileIoUtils {
     private static final Log LOG = Log.getInstance(BamFileIoUtils.class);
 
-    public static final String BAM_FILE_EXTENSION = ".bam";
+    public static final String BAM_FILE_EXTENSION = "." + SamReader.Type.BAM_TYPE.fileExtension();
 
     public static boolean isBamFile(final File file) {
-        return ((file != null) && file.getName().endsWith(BAM_FILE_EXTENSION));
+        return ((file != null) && SamReader.Type.BAM_TYPE.hasValidFileExtension(file.getName()));
```

Comment:
You could use an import to make it a bit more readable: `import static htsjdk.samtools.SamReader.Type.*;`.  Your call.

Reply:
I think that the accepted view is against static imports...